parsers/uniprot/xml.go

```diff
@@ -0,0 +1,585 @@
+	return e.EncodeElement(layout, start)
+}
+func (t *SequenceType) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	type T SequenceType
+	var overlay struct {
+		*T
+		Modified *xsdDate `xml:"modified,attr"`
+	}
+	overlay.T = (*T)(t)
+	overlay.Modified = (*xsdDate)(&overlay.T.Modified)
+	return d.DecodeElement(&overlay, &start)
+}
+
+// Describes the source of the sequence according to the citation.
+// Equivalent to the flat file RC-line.
+type SourceDataType struct {
+	Strain     Strain     `xml:"http://uniprot.org/uniprot strain,omitempty"`
+	Plasmid    Plasmid    `xml:"http://uniprot.org/uniprot plasmid,omitempty"`
+	Transposon Transposon `xml:"http://uniprot.org/uniprot transposon,omitempty"`
+	Tissue     Tissue     `xml:"http://uniprot.org/uniprot tissue,omitempty"`
+}
+
+// Describes the source of the data using a database cross-reference (or a 'ref' attribute when the source cannot be found in a public data source, such as PubMed, and is cited only within the UniProtKB entry).
+type SourceType struct {
+	DbReference DbReferenceType `xml:"http://uniprot.org/uniprot dbReference,omitempty"`
+}
+
+// May be one of certain, uncertain, less than, greater than, unknown
+type Status string
+
+// Indicates whether the name of a plasmid is known or unknown.
+type StatusType struct {
+	Value  string `xml:",chardata"`
+	Status Status `xml:"status,attr,omitempty"`
+}
+
+func (t *StatusType) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	type T StatusType
+	var overlay struct {
+		*T
+		Status *Status `xml:"status,attr,omitempty"`
+	}
+	overlay.T = (*T)(t)
+	overlay.Status = (*Status)(&overlay.T.Status)
+	return d.DecodeElement(&overlay, &start)
+}
+
+type Strain struct {
+	Value    string      `xml:",chardata"`
+	Evidence IntListType `xml:"evidence,attr,omitempty"`
+}
+
+// Describes the subcellular location and optionally the topology and orientation of a molecule.
+type SubcellularLocationType struct {
+	Location    []EvidencedStringType `xml:"http://uniprot.org/uniprot location"`
+	Topology    []EvidencedStringType `xml:"http://uniprot.org/uniprot topology,omitempty"`
+	Orientation []EvidencedStringType `xml:"http://uniprot.org/uniprot orientation,omitempty"`
+}
+
+type SubmittedName struct {
+	FullName EvidencedStringType   `xml:"http://uniprot.org/uniprot fullName"`
+	EcNumber []EvidencedStringType `xml:"http://uniprot.org/uniprot ecNumber,omitempty"`
+}
+
+type TemperatureDependence struct {
+	Text []EvidencedStringType `xml:"http://uniprot.org/uniprot text"`
+}
+
+type Tissue struct {
+	Value    string      `xml:",chardata"`
+	Evidence IntListType `xml:"evidence,attr,omitempty"`
+}
+
+type Transposon struct {
+	Value    string      `xml:",chardata"`
+	Evidence IntListType `xml:"evidence,attr,omitempty"`
+}
+
+// May be one of apicoplast, chloroplast, organellar chromatophore, cyanelle, hydrogenosome, mitochondrion, non-photosynthetic plastid, nucleomorph, plasmid, plastid
+type Type string
```

Comment:
This needs a more descriptive name.

Reply:
Oh just reread the comment and this is just autogenerated code? 🤔 

Is it important that we adhere strictly to the autogeneration or would fixing up some of the naming be reasonable?

Reply:
We could fix it up, but we might have to autogenerate again, in which case all of our edits would be wiped. So I think it is better to document outside the autogenerated stuff.